# 🛒 E-Commerce Sales Analysis Project

## İstatistik & Python Bootcamp — Final Projesi

Bu notebook, bootcamp boyunca öğrendiğimiz konuları tek bir veri seti üzerinde uygulamak için hazırlanmıştır.

Bu projede bir e-ticaret sitesinin sipariş verilerini analiz edeceğiz.

## Proje Amacı

Bu çalışmanın amacı:

- CSV dosyasını bilgisayardan Colab'e yüklemek
- Pandas ile veri setini okumak
- Veri setini tanımak
- Özet istatistikleri yorumlamak
- Filtreleme yapmak
- Segmentlere göre analiz yapmak
- İş dünyasına uygun içgörüler üretmek
- GitHub'a yüklenebilecek bir portfolyo projesi hazırlamak

## Kullanacağımız Veri Seti

Veri seti örnek bir e-ticaret sipariş datasıdır.

İçerisinde şu bilgiler yer alır:

- siparis_id
- siparis_tarihi
- musteri_adi
- sehir
- kategori
- urun
- uyelik_tipi
- odeme_yontemi
- urun_fiyati
- adet
- indirim_orani
- toplam_tutar
- kargo_suresi_gun
- siparis_durumu

# 1. Adım — Kütüphaneleri Çağırma

Pandas, Python'da veri analizi yapmak için en sık kullanılan kütüphanelerden biridir.

Bu projede veriyi tablo gibi okumak, incelemek, filtrelemek ve analiz etmek için Pandas kullanacağız.

In [30]:
import pandas as pd

# 2. Adım — CSV Dosyasını Bilgisayardan Yükleme

Bu adımda CSV dosyasını bilgisayarımızdan Colab'e yükleyeceğiz.

Kod çalışınca **Choose Files / Dosya Seç** butonu çıkacak.  
Buradan `ecommerce_sales_final_project.csv` dosyasını seçmelisin.

In [31]:
from google.colab import files

uploaded = files.upload()

Saving dataset_ecommerce_sales_final_project.csv to dataset_ecommerce_sales_final_project (2).csv


# 3. Adım — CSV Dosyasını Okuma

Dosyayı yükledikten sonra Pandas ile okuyacağız.

> Not: Dosya adının doğru yazıldığından emin ol.

In [32]:
df = pd.read_csv("dataset_ecommerce_sales_final_project.csv")

df

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
0,ORD-0001,2026-05-03,Ali,Trabzon,Kitap,Roman,Standart,Kredi Kartı,470,1,0.05,446.5,5,Tamamlandı
1,ORD-0002,2026-06-07,Ceren,Konya,Moda,Gömlek,Silver,Kredi Kartı,1180,2,0.05,2242.0,3,Tamamlandı
2,ORD-0003,2026-05-25,Zeynep,Eskişehir,Spor,Dambıl Seti,Standart,Havale,1810,1,0.00,1810.0,3,Tamamlandı
3,ORD-0004,2026-05-19,Tuna,Konya,Oyuncak,Araba Seti,Silver,Havale,490,1,0.10,441.0,5,Tamamlandı
4,ORD-0005,2026-05-25,Ceren,Bursa,Ev & Yaşam,Masa Lambası,Standart,Kapıda Ödeme,260,1,0.05,247.0,2,Tamamlandı
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
95,ORD-0096,2026-06-07,Derya,Adana,Market,Bitki Çayı,Standart,Havale,320,3,0.00,960.0,2,Tamamlandı
96,ORD-0097,2026-06-10,Tuna,Muğla,Elektronik,Akıllı Saat,Gold,Kapıda Ödeme,1890,2,0.10,3402.0,3,Tamamlandı
97,ORD-0098,2026-06-26,Emre,Adana,Market,Zeytinyağı,Standart,Banka Kartı,540,1,0.05,513.0,4,Tamamlandı
98,ORD-0099,2026-06-11,Zeynep,İzmir,Kozmetik,Nemlendirici,Gold,Kapıda Ödeme,850,2,0.10,1530.0,3,Tamamlandı


# 4. Adım — Veri Setini Tanıma (EDA)

EDA, yani **Exploratory Data Analysis**, veriyi tanıma aşamasıdır.

Bir veri analisti analize başlamadan önce şu soruları sorar:

- Kaç satır var?
- Kaç sütun var?
- Hangi kolonlar var?
- Veri tipleri doğru mu?
- Eksik veri var mı?
- Sayısal kolonlar ne anlatıyor?

## İlk Satırları Görüntüleme

`head()` veri setinin ilk 5 satırını gösterir.

Bu, verinin doğru yüklenip yüklenmediğini kontrol etmek için kullanılır.

In [33]:
df.head(30)

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
0,ORD-0001,2026-05-03,Ali,Trabzon,Kitap,Roman,Standart,Kredi Kartı,470,1,0.05,446.5,5,Tamamlandı
1,ORD-0002,2026-06-07,Ceren,Konya,Moda,Gömlek,Silver,Kredi Kartı,1180,2,0.05,2242.0,3,Tamamlandı
2,ORD-0003,2026-05-25,Zeynep,Eskişehir,Spor,Dambıl Seti,Standart,Havale,1810,1,0.00,1810.0,3,Tamamlandı
3,ORD-0004,2026-05-19,Tuna,Konya,Oyuncak,Araba Seti,Silver,Havale,490,1,0.10,441.0,5,Tamamlandı
4,ORD-0005,2026-05-25,Ceren,Bursa,Ev & Yaşam,Masa Lambası,Standart,Kapıda Ödeme,260,1,0.05,247.0,2,Tamamlandı
5,ORD-0006,2026-05-11,Emre,Ankara,Kitap,Kişisel Gelişim Kitabı,Silver,Banka Kartı,160,1,0.05,152.0,2,Tamamlandı
6,ORD-0007,2026-06-19,Ayşe,Eskişehir,Elektronik,Powerbank,Silver,Banka Kartı,1270,4,0.05,4826.0,5,Tamamlandı
7,ORD-0008,2026-06-01,Derya,Eskişehir,Oyuncak,Peluş Oyuncak,Gold,Kapıda Ödeme,670,1,0.10,603.0,4,Tamamlandı
8,ORD-0009,2026-05-09,İrem,Muğla,Elektronik,Akıllı Saat,Standart,Kapıda Ödeme,1200,2,0.00,2400.0,3,Tamamlandı
9,ORD-0010,2026-06-08,Buse,Muğla,Elektronik,Bluetooth Kulaklık,Standart,Havale,3080,1,0.00,3080.0,5,Tamamlandı


## Son Satırları Görüntüleme

`tail()` veri setinin son 5 satırını gösterir.

In [34]:
df.tail(20)

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
80,ORD-0081,2026-06-25,Can,Muğla,Moda,Gömlek,Silver,Kapıda Ödeme,660,1,0.05,627.0,4,İptal
81,ORD-0082,2026-05-16,Aslı,Bursa,Ev & Yaşam,Kupa Seti,Standart,Kapıda Ödeme,790,3,0.00,2370.0,3,İptal
82,ORD-0083,2026-06-20,Tuna,Konya,Moda,Çanta,Silver,Banka Kartı,320,2,0.05,608.0,4,İptal
83,ORD-0084,2026-05-01,Arda,Adana,Ev & Yaşam,Masa Lambası,Standart,Banka Kartı,410,4,0.05,1558.0,4,Tamamlandı
84,ORD-0085,2026-06-24,Selin,Konya,Kozmetik,Parfüm,Standart,Kapıda Ödeme,230,3,0.00,690.0,2,Tamamlandı
85,ORD-0086,2026-06-17,Mehmet,Antalya,Market,Granola,Standart,Kapıda Ödeme,700,1,0.00,700.0,2,İptal
86,ORD-0087,2026-05-24,Kerem,İstanbul,Market,Kahve,Gold,Banka Kartı,380,2,0.15,646.0,4,Tamamlandı
87,ORD-0088,2026-05-16,Onur,Konya,Moda,Çanta,Standart,Banka Kartı,1130,3,0.05,3220.5,3,Tamamlandı
88,ORD-0089,2026-05-11,Mehmet,Antalya,Moda,Ceket,Gold,Kapıda Ödeme,960,2,0.10,1728.0,2,Tamamlandı
89,ORD-0090,2026-05-25,Eylül,Antalya,Market,Bitki Çayı,Silver,Kapıda Ödeme,270,4,0.05,1026.0,3,Tamamlandı


## Satır ve Sütun Sayısını Görme

`shape` bize veri setinin boyutunu verir.

İlk değer satır sayısı, ikinci değer sütun sayısıdır.

In [35]:
df.shape

(100, 14)

## Kolon İsimlerini Görme

`columns` veri setindeki sütun isimlerini gösterir.

In [36]:
df.columns

Index(['siparis_id', 'siparis_tarihi', 'musteri_adi', 'sehir', 'kategori',
       'urun', 'uyelik_tipi', 'odeme_yontemi', 'urun_fiyati', 'adet',
       'indirim_orani', 'toplam_tutar', 'kargo_suresi_gun', 'siparis_durumu'],
      dtype='object')

## Veri Tiplerini ve Eksik Verileri Kontrol Etme

`info()` bize şunları gösterir:

- Her kolonda kaç veri var?
- Veri tipi nedir?
- Eksik veri var mı?

In [37]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 14 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   siparis_id        100 non-null    object 
 1   siparis_tarihi    100 non-null    object 
 2   musteri_adi       100 non-null    object 
 3   sehir             100 non-null    object 
 4   kategori          100 non-null    object 
 5   urun              100 non-null    object 
 6   uyelik_tipi       100 non-null    object 
 7   odeme_yontemi     100 non-null    object 
 8   urun_fiyati       100 non-null    int64  
 9   adet              100 non-null    int64  
 10  indirim_orani     100 non-null    float64
 11  toplam_tutar      100 non-null    float64
 12  kargo_suresi_gun  100 non-null    int64  
 13  siparis_durumu    100 non-null    object 
dtypes: float64(2), int64(3), object(9)
memory usage: 11.1+ KB


# 5. Adım — Özet İstatistikler

`describe()` sayısal kolonlar için hızlı bir istatistik özeti verir.

Bu komut bize şunları gösterir:

- count: veri sayısı
- mean: ortalama
- std: standart sapma
- min: minimum değer
- 25%: birinci çeyrek
- 50%: medyan
- 75%: üçüncü çeyrek
- max: maksimum değer

İlk hafta öğrendiğimiz istatistik kavramlarını burada tek bir komutla görebiliyoruz.

## Describe Çıktısını Yorumlama

Özellikle şu kolonlara dikkat edebiliriz:

- `toplam_tutar`: Siparişlerin parasal büyüklüğü
- `adet`: Siparişte kaç ürün olduğu
- `indirim_orani`: İndirim oranı
- `kargo_suresi_gun`: Kargo süresi

Yorumlarken şu soruları sor:

- Ortalama sipariş tutarı kaç?
- Minimum ve maksimum sipariş tutarı arasında büyük fark var mı?
- Standart sapma yüksek mi?
- Kargo süresi genel olarak tutarlı mı?

In [38]:
df.describe()

,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun
count,100.000000,100.000000,100.000000,100.000000,100.000000
mean,931.100000,2.000000,0.046000,1758.780000,2.920000
std,736.343539,1.214392,0.041243,1854.705844,1.203362
min,100.000000,1.000000,0.000000,100.000000,1.000000
25%,440.000000,1.000000,0.000000,606.750000,2.000000
50%,695.000000,1.500000,0.050000,1173.000000,3.000000
75%,1185.000000,3.000000,0.050000,2240.500000,4.000000
max,3410.000000,5.000000,0.150000,10800.000000,7.000000


* Veri setinde toplam 100 sipariş bulunmaktadır.
* Ortalama ürün fiyatı 931 TL, ortalama sipariş tutarı ise 1.759 TL’dir.
* Bir siparişte ortalama 2 adet ürün satın alınmıştır.
* Ortalama indirim oranı %4,6’dır. Genel olarak yüksek indirimler uygulanmamıştır.
* Ortalama kargo süresi yaklaşık 3 gündür, bu da siparişlerin çoğunun kısa sürede teslim edildiğini göstermektedir.
* En düşük sipariş tutarı 100 TL, en yüksek sipariş tutarı ise 10.800 TL’dir. Bu durum veri setinde düşük ve yüksek tutarlı siparişlerin birlikte bulunduğunu göstermektedir.
* Ürün fiyatlarının standart sapması (736 TL) ve toplam sipariş tutarının standart sapması (1.855 TL) yüksektir. Bu da müşteri harcamalarının birbirinden oldukça farklı olduğunu göstermektedir.
* Siparişlerin %50’sinde (medyan) toplam tutar 1.173 TL veya daha düşüktür.
* Siparişlerin %75’i 2.240 TL ve altındadır. Sadece en yüksek harcama yapan %25’lik grup bu değerin üzerinde sipariş vermiştir.

⸻

🎯 Genel İçgörü

* Ortalama sipariş tutarı yaklaşık 1.759 TL’dir.
* Müşteri harcamaları arasında belirgin farklılıklar bulunmaktadır.
* Kargo süreleri genel olarak kısa olsa da sipariş tutarları geniş bir aralığa yayılmıştır.
* Veri seti, farklı müşteri segmentlerinin bulunduğunu göstermektedir ve segment bazlı analizler yapmak için uygundur.

# 6. Adım — Temel Metrikleri Hesaplama

Şimdi iş tarafında anlamlı olabilecek bazı metrikleri hesaplayalım.

## Ortalama Sipariş Tutarı

In [39]:
df["toplam_tutar"].mean()

np.float64(1758.78)

## Medyan Sipariş Tutarı

Medyan, özellikle yüksek tutarlı birkaç sipariş ortalamayı yukarı çekiyorsa daha anlamlı olabilir.

In [40]:
df["toplam_tutar"].median()

1173.0

## En Düşük ve En Yüksek Sipariş Tutarı

In [41]:
print("En düşük sipariş tutarı:", df["toplam_tutar"].min())
print("En yüksek sipariş tutarı:", df["toplam_tutar"].max())

En düşük sipariş tutarı: 100.0
En yüksek sipariş tutarı: 10800.0


## Toplam Ciro

Burada sadece `Tamamlandı` durumundaki siparişleri dikkate alıyoruz.

Çünkü iptal veya iade edilen siparişleri gerçek gelir gibi değerlendirmek yanıltıcı olabilir.

In [42]:
tamamlanan_siparisler = df[df["siparis_durumu"] == "Tamamlandı"]

toplam_ciro = tamamlanan_siparisler["toplam_tutar"].sum()

print("Toplam Ciro:", toplam_ciro)

Toplam Ciro: 158271.5


## Tamamlanan Sipariş Oranı

In [43]:
tamamlanma_orani = (df["siparis_durumu"] == "Tamamlandı").mean() * 100

print("Tamamlanan Sipariş Oranı:", tamamlanma_orani)

Tamamlanan Sipariş Oranı: 89.0


# 7. Adım — Filtreleme

Filtreleme, veri analistlerinin en sık yaptığı işlemlerden biridir.

Belirli bir koşula uyan satırları görmek için kullanılır.

## 1000 TL Üzeri Siparişler

In [44]:
df[df["toplam_tutar"] > 1000]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
1,ORD-0002,2026-06-07,Ceren,Konya,Moda,Gömlek,Silver,Kredi Kartı,1180,2,0.05,2242.0,3,Tamamlandı
2,ORD-0003,2026-05-25,Zeynep,Eskişehir,Spor,Dambıl Seti,Standart,Havale,1810,1,0.00,1810.0,3,Tamamlandı
6,ORD-0007,2026-06-19,Ayşe,Eskişehir,Elektronik,Powerbank,Silver,Banka Kartı,1270,4,0.05,4826.0,5,Tamamlandı
8,ORD-0009,2026-05-09,İrem,Muğla,Elektronik,Akıllı Saat,Standart,Kapıda Ödeme,1200,2,0.00,2400.0,3,Tamamlandı
9,ORD-0010,2026-06-08,Buse,Muğla,Elektronik,Bluetooth Kulaklık,Standart,Havale,3080,1,0.00,3080.0,5,Tamamlandı
11,ORD-0012,2026-06-02,Mina,Antalya,Elektronik,Telefon Kılıfı,Silver,Banka Kartı,3410,3,0.10,9207.0,2,Tamamlandı
12,ORD-0013,2026-05-08,Barış,Eskişehir,Elektronik,Akıllı Saat,Silver,Havale,2100,3,0.10,5670.0,3,Tamamlandı
16,ORD-0017,2026-06-01,Burak,Bursa,Elektronik,Bluetooth Kulaklık,Gold,Banka Kartı,1150,4,0.15,3910.0,1,Tamamlandı
17,ORD-0018,2026-05-28,Efe,Ankara,Spor,Su Matarası,Silver,Kapıda Ödeme,1530,1,0.05,1453.5,3,Tamamlandı
18,ORD-0019,2026-05-07,Deniz,Eskişehir,Kozmetik,Güneş Kremi,Standart,Banka Kartı,860,5,0.00,4300.0,2,Tamamlandı


## 5000 TL Üzeri Yüksek Tutarli Siparişler

In [45]:
df[df["toplam_tutar"] > 5000]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
11,ORD-0012,2026-06-02,Mina,Antalya,Elektronik,Telefon Kılıfı,Silver,Banka Kartı,3410,3,0.10,9207.0,2,Tamamlandı
12,ORD-0013,2026-05-08,Barış,Eskişehir,Elektronik,Akıllı Saat,Silver,Havale,2100,3,0.10,5670.0,3,Tamamlandı
20,ORD-0021,2026-05-31,Burak,İzmir,Elektronik,Kablosuz Mouse,Gold,Kapıda Ödeme,3000,4,0.10,10800.0,4,Tamamlandı
53,ORD-0054,2026-05-07,Aslı,İzmir,Spor,Dambıl Seti,Silver,Kredi Kartı,1590,4,0.05,6042.0,3,İade
55,ORD-0056,2026-05-28,Kaan,İstanbul,Elektronik,Telefon Kılıfı,Standart,Havale,3040,2,0.00,6080.0,4,Tamamlandı
71,ORD-0072,2026-05-18,İrem,Adana,Elektronik,Telefon Kılıfı,Gold,Kapıda Ödeme,2260,3,0.10,6102.0,7,Tamamlandı


## Sadece Premium / Gold Üyeler

Bu veri setinde üyelik tipleri:

- Standart
- Silver
- Gold

In [46]:
df[df["uyelik_tipi"] == "Gold"]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
7,ORD-0008,2026-06-01,Derya,Eskişehir,Oyuncak,Peluş Oyuncak,Gold,Kapıda Ödeme,670,1,0.10,603.0,4,Tamamlandı
10,ORD-0011,2026-05-28,Elif,Eskişehir,Market,Kahve,Gold,Kapıda Ödeme,560,1,0.10,504.0,2,Tamamlandı
16,ORD-0017,2026-06-01,Burak,Bursa,Elektronik,Bluetooth Kulaklık,Gold,Banka Kartı,1150,4,0.15,3910.0,1,Tamamlandı
20,ORD-0021,2026-05-31,Burak,İzmir,Elektronik,Kablosuz Mouse,Gold,Kapıda Ödeme,3000,4,0.10,10800.0,4,Tamamlandı
26,ORD-0027,2026-06-12,Tuna,Muğla,Market,Atıştırmalık,Gold,Havale,330,1,0.10,297.0,1,İptal
28,ORD-0029,2026-05-03,Ali,Konya,Moda,Ceket,Gold,Havale,1030,1,0.15,875.5,2,Tamamlandı
41,ORD-0042,2026-06-26,Yasemin,Ankara,Moda,Tişört,Gold,Banka Kartı,480,1,0.10,432.0,3,Tamamlandı
49,ORD-0050,2026-06-07,Aslı,Trabzon,Moda,Çanta,Gold,Kredi Kartı,1370,3,0.10,3699.0,3,Tamamlandı
57,ORD-0058,2026-05-21,Eylül,Adana,Kozmetik,Şampuan,Gold,Banka Kartı,1160,1,0.10,1044.0,3,Tamamlandı
71,ORD-0072,2026-05-18,İrem,Adana,Elektronik,Telefon Kılıfı,Gold,Kapıda Ödeme,2260,3,0.10,6102.0,7,Tamamlandı


## Elektronik Kategorisindeki Siparişler

In [47]:
df[df["kategori"] == "Elektronik"]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
6,ORD-0007,2026-06-19,Ayşe,Eskişehir,Elektronik,Powerbank,Silver,Banka Kartı,1270,4,0.05,4826.0,5,Tamamlandı
8,ORD-0009,2026-05-09,İrem,Muğla,Elektronik,Akıllı Saat,Standart,Kapıda Ödeme,1200,2,0.00,2400.0,3,Tamamlandı
9,ORD-0010,2026-06-08,Buse,Muğla,Elektronik,Bluetooth Kulaklık,Standart,Havale,3080,1,0.00,3080.0,5,Tamamlandı
11,ORD-0012,2026-06-02,Mina,Antalya,Elektronik,Telefon Kılıfı,Silver,Banka Kartı,3410,3,0.10,9207.0,2,Tamamlandı
12,ORD-0013,2026-05-08,Barış,Eskişehir,Elektronik,Akıllı Saat,Silver,Havale,2100,3,0.10,5670.0,3,Tamamlandı
16,ORD-0017,2026-06-01,Burak,Bursa,Elektronik,Bluetooth Kulaklık,Gold,Banka Kartı,1150,4,0.15,3910.0,1,Tamamlandı
20,ORD-0021,2026-05-31,Burak,İzmir,Elektronik,Kablosuz Mouse,Gold,Kapıda Ödeme,3000,4,0.10,10800.0,4,Tamamlandı
43,ORD-0044,2026-06-18,Gizem,Trabzon,Elektronik,Akıllı Saat,Standart,Kapıda Ödeme,3250,1,0.00,3250.0,3,İptal
46,ORD-0047,2026-05-27,Ayşe,Eskişehir,Elektronik,Akıllı Saat,Standart,Banka Kartı,1620,1,0.00,1620.0,2,Tamamlandı
50,ORD-0051,2026-05-22,Melis,Bursa,Elektronik,Powerbank,Standart,Kapıda Ödeme,860,1,0.00,860.0,3,Tamamlandı


## Tamamlanan Siparişler

In [48]:
df[df["siparis_durumu"] == "Tamamlandı"]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
0,ORD-0001,2026-05-03,Ali,Trabzon,Kitap,Roman,Standart,Kredi Kartı,470,1,0.05,446.5,5,Tamamlandı
1,ORD-0002,2026-06-07,Ceren,Konya,Moda,Gömlek,Silver,Kredi Kartı,1180,2,0.05,2242.0,3,Tamamlandı
2,ORD-0003,2026-05-25,Zeynep,Eskişehir,Spor,Dambıl Seti,Standart,Havale,1810,1,0.00,1810.0,3,Tamamlandı
3,ORD-0004,2026-05-19,Tuna,Konya,Oyuncak,Araba Seti,Silver,Havale,490,1,0.10,441.0,5,Tamamlandı
4,ORD-0005,2026-05-25,Ceren,Bursa,Ev & Yaşam,Masa Lambası,Standart,Kapıda Ödeme,260,1,0.05,247.0,2,Tamamlandı
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
95,ORD-0096,2026-06-07,Derya,Adana,Market,Bitki Çayı,Standart,Havale,320,3,0.00,960.0,2,Tamamlandı
96,ORD-0097,2026-06-10,Tuna,Muğla,Elektronik,Akıllı Saat,Gold,Kapıda Ödeme,1890,2,0.10,3402.0,3,Tamamlandı
97,ORD-0098,2026-06-26,Emre,Adana,Market,Zeytinyağı,Standart,Banka Kartı,540,1,0.05,513.0,4,Tamamlandı
98,ORD-0099,2026-06-11,Zeynep,İzmir,Kozmetik,Nemlendirici,Gold,Kapıda Ödeme,850,2,0.10,1530.0,3,Tamamlandı


## Birden Fazla Koşulla Filtreleme

Örnek soru:

> Gold üyelerin 1000 TL üzerindeki siparişleri hangileri?

In [49]:
df[
    (df["uyelik_tipi"] == "Gold")
    &
    (df["toplam_tutar"] > 1000)
]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
16,ORD-0017,2026-06-01,Burak,Bursa,Elektronik,Bluetooth Kulaklık,Gold,Banka Kartı,1150,4,0.15,3910.0,1,Tamamlandı
20,ORD-0021,2026-05-31,Burak,İzmir,Elektronik,Kablosuz Mouse,Gold,Kapıda Ödeme,3000,4,0.10,10800.0,4,Tamamlandı
49,ORD-0050,2026-06-07,Aslı,Trabzon,Moda,Çanta,Gold,Kredi Kartı,1370,3,0.10,3699.0,3,Tamamlandı
57,ORD-0058,2026-05-21,Eylül,Adana,Kozmetik,Şampuan,Gold,Banka Kartı,1160,1,0.10,1044.0,3,Tamamlandı
71,ORD-0072,2026-05-18,İrem,Adana,Elektronik,Telefon Kılıfı,Gold,Kapıda Ödeme,2260,3,0.10,6102.0,7,Tamamlandı
78,ORD-0079,2026-05-18,Kerem,Antalya,Elektronik,Bluetooth Kulaklık,Gold,Kapıda Ödeme,1860,1,0.10,1674.0,3,Tamamlandı
88,ORD-0089,2026-05-11,Mehmet,Antalya,Moda,Ceket,Gold,Kapıda Ödeme,960,2,0.10,1728.0,2,Tamamlandı
90,ORD-0091,2026-06-06,Eylül,Konya,Elektronik,Powerbank,Gold,Kredi Kartı,2190,2,0.10,3942.0,3,Tamamlandı
94,ORD-0095,2026-06-18,Deniz,Adana,Moda,Çanta,Gold,Kapıda Ödeme,1750,1,0.10,1575.0,3,İade
96,ORD-0097,2026-06-10,Tuna,Muğla,Elektronik,Akıllı Saat,Gold,Kapıda Ödeme,1890,2,0.10,3402.0,3,Tamamlandı


# 8. Adım — Segmentasyon ve Gruplama

Segmentasyon, veriyi anlamlı gruplara ayırmak demektir.

Örneğin:

- Şehre göre analiz
- Kategoriye göre analiz
- Üyelik tipine göre analiz
- Sipariş durumuna göre analiz

Bunun için `groupby()` kullanırız.

## Üyelik Tipine Göre Ortalama Sipariş Tutarı

In [50]:
df.groupby("uyelik_tipi")["toplam_tutar"].mean()

,toplam_tutar
uyelik_tipi,
Gold,2515.500000
Silver,1743.305556
Standart,1496.925532


## Kategoriye Göre Toplam Ciro

In [51]:
tamamlanan_siparisler.groupby("kategori")["toplam_tutar"].sum().sort_values(ascending=False)

,toplam_tutar
kategori,
Elektronik,67428.5
Moda,23174.0
Kozmetik,22029.0
Market,18591.0
Oyuncak,8618.5
Kitap,7126.5
Spor,6948.0
Ev & Yaşam,4356.0


## Şehre Göre Toplam Ciro

In [52]:
tamamlanan_siparisler.groupby("sehir")["toplam_tutar"].sum().sort_values(ascending=False)

,toplam_tutar
sehir,
Eskişehir,25195.5
Adana,21680.5
İzmir,19130.0
Antalya,19015.0
Konya,18021.5
Muğla,15341.0
Ankara,12969.5
Bursa,11071.0
İstanbul,9951.5


## Sipariş Durumuna Göre Sipariş Sayısı

In [53]:
df["siparis_durumu"].value_counts()

,count
siparis_durumu,
Tamamlandı,89
İptal,8
İade,3


## Kategoriye Göre Ortalama Kargo Süresi

In [54]:
df.groupby("kategori")["kargo_suresi_gun"].mean().sort_values(ascending=False)

,kargo_suresi_gun
kategori,
Oyuncak,3.555556
Elektronik,3.277778
Kitap,3.000000
Ev & Yaşam,3.000000
Market,2.750000
Kozmetik,2.714286
Moda,2.687500
Spor,2.500000


#Bonus: Korelasyon
**df.corr()**

Korelasyon, sayılar arasındaki ilişkiyi ölçer.

*Korelasyon Analizi*

Korelasyon katsayısı -1 ile +1 arasında değer alır.

* +1’e yaklaştıkça: Güçlü pozitif ilişki
* 0’a yaklaştıkça: İlişki yok veya çok zayıf
* -1’e yaklaştıkça: Güçlü negatif ilişki

Veri setimizde Order_ID, Şehir, Kategori ve Üyelik Tipi gibi metinsel sütunlar bulunduğu için önce sayısal sütunları seçtik, ardından korelasyon analizini gerçekleştirdik.

In [55]:
df.select_dtypes(include="number").corr()

,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun
urun_fiyati,1.000000,-0.000452,0.132856,0.764104,0.135641
adet,-0.000452,1.000000,-0.030251,0.513181,0.124418
indirim_orani,0.132856,-0.030251,1.000000,0.169830,0.034192
toplam_tutar,0.764104,0.513181,0.169830,1.000000,0.166860
kargo_suresi_gun,0.135641,0.124418,0.034192,0.166860,1.000000


Yukarıdaki korelasyon tablosuna göre aşağıdaki çıkarımları yapabiliriz:

1. Ürün Fiyatı ↔ Toplam Tutar (0.76)

Ürün fiyatı ile toplam sipariş tutarı arasında güçlü pozitif ilişki bulunmaktadır.

Bu, ürün fiyatı arttıkça toplam sipariş tutarının da artma eğiliminde olduğunu göstermektedir.

⸻

2. Adet ↔ Toplam Tutar (0.51)

Sipariş edilen ürün adedi ile toplam sipariş tutarı arasında orta seviyede pozitif ilişki bulunmaktadır.

Müşteriler daha fazla ürün satın aldıkça toplam harcama da artmaktadır.

⸻

3. Ürün Fiyatı ↔ Adet (-0.00)

Bu iki değişken arasında anlamlı bir ilişki bulunmamaktadır.

Yani pahalı ürünlerin daha az veya daha fazla adetle satın alındığını bu veri setine göre söyleyemeyiz.

⸻

4. İndirim Oranı ↔ Toplam Tutar (0.17)

İndirim oranı ile toplam sipariş tutarı arasında çok zayıf pozitif ilişki bulunmaktadır.

Bu veri setinde indirim oranı, müşterilerin toplam harcamasını belirleyen güçlü bir faktör olarak görünmemektedir.

⸻

5. Kargo Süresi ↔ Toplam Tutar (0.17)

Kargo süresi ile toplam sipariş tutarı arasında çok zayıf bir ilişki bulunmaktadır.

Bu sonuç, yüksek tutarlı siparişlerin daha uzun sürede teslim edildiğini göstermemektedir.

⸻

🎯 Genel Değerlendirme

Bu korelasyon analizine göre:

* En güçlü ilişki, ürün fiyatı ile toplam sipariş tutarı arasındadır.
* Sipariş adedi de toplam harcamayı etkileyen önemli değişkenlerden biridir.
* İndirim oranı ve kargo süresi değişkenlerinin toplam sipariş tutarı ile güçlü bir ilişkisi bulunmamaktadır.

Not: Korelasyon, iki değişken arasındaki ilişkinin gücünü gösterir. Tek başına bir değişkenin diğerine neden olduğunu kanıtlamaz. Bu nedenle korelasyon sonuçları yorumlanırken her zaman iş bilgisiyle birlikte değerlendirilmelidir.

# 9. Adım — İş Sorularını Cevaplama

Şimdi veri analisti gibi bazı iş sorularını cevaplayalım.

## Soru 1 — En yüksek tutarlı sipariş hangisi?

In [56]:
df[df["toplam_tutar"] == df["toplam_tutar"].max()]

,siparis_id,siparis_tarihi,musteri_adi,sehir,kategori,urun,uyelik_tipi,odeme_yontemi,urun_fiyati,adet,indirim_orani,toplam_tutar,kargo_suresi_gun,siparis_durumu
20,ORD-0021,2026-05-31,Burak,İzmir,Elektronik,Kablosuz Mouse,Gold,Kapıda Ödeme,3000,4,0.1,10800.0,4,Tamamlandı


## Soru 2 — En çok ciro getiren kategori hangisi?

In [57]:
tamamlanan_siparisler.groupby("kategori")["toplam_tutar"].sum().sort_values(ascending=False).head(1)

,toplam_tutar
kategori,
Elektronik,67428.5


## Soru 3 — Gold müşteriler daha yüksek harcama yapıyor mu?

In [58]:
df.groupby("uyelik_tipi")["toplam_tutar"].mean().sort_values(ascending=False)

,toplam_tutar
uyelik_tipi,
Gold,2515.500000
Silver,1743.305556
Standart,1496.925532


## Soru 4 — En çok ciro getiren şehir hangisi?

In [59]:
tamamlanan_siparisler.groupby("sehir")["toplam_tutar"].sum().sort_values(ascending=False).head(1)

,toplam_tutar
sehir,
Eskişehir,25195.5


## Soru 5 — Ortalama kargo süresi kaç gün?

In [60]:
df["kargo_suresi_gun"].mean()

np.float64(2.92)

# 10. Adım — Veri Analisti Gibi İçgörü Yazma

Veri analizi sadece kod yazmak değildir.

Bir veri analisti bulduğu sonuçları iş dünyasına uygun şekilde yorumlamalıdır.

Aşağıdaki hücrelerde bu veri setinden çıkarılabilecek örnek içgörüler yer alıyor.

In [61]:
print("İçgörü 1: Ortalama sipariş tutarı yaklaşık 1759 TL'dir. Ancak medyan 1173 TL olduğu için bazı yüksek tutarlı siparişler ortalamayı yukarı çekiyor olabilir.")

print("İçgörü 2: Tamamlanan siparişlerden elde edilen toplam ciro yaklaşık 158272 TL'dir.")

print("İçgörü 3: En çok ciro getiren kategori Elektronik kategorisidir. Bu kategori satış stratejisinde önceliklendirilebilir.")

print("İçgörü 4: Gold üyelerin ortalama sipariş tutarı Standart üyelerden daha yüksektir. Bu durum sadakat programlarının harcama davranışını etkileyebileceğini gösterir.")

print("İçgörü 5: Standart sapmanın yüksek olması sipariş tutarlarının birbirinden oldukça farklı olduğunu gösterir. Bu nedenle müşterileri düşük, orta ve yüksek harcama yapan gruplara ayırmak faydalı olabilir.")

İçgörü 1: Ortalama sipariş tutarı yaklaşık 1759 TL'dir. Ancak medyan 1173 TL olduğu için bazı yüksek tutarlı siparişler ortalamayı yukarı çekiyor olabilir.
İçgörü 2: Tamamlanan siparişlerden elde edilen toplam ciro yaklaşık 158272 TL'dir.
İçgörü 3: En çok ciro getiren kategori Elektronik kategorisidir. Bu kategori satış stratejisinde önceliklendirilebilir.
İçgörü 4: Gold üyelerin ortalama sipariş tutarı Standart üyelerden daha yüksektir. Bu durum sadakat programlarının harcama davranışını etkileyebileceğini gösterir.
İçgörü 5: Standart sapmanın yüksek olması sipariş tutarlarının birbirinden oldukça farklı olduğunu gösterir. Bu nedenle müşterileri düşük, orta ve yüksek harcama yapan gruplara ayırmak faydalı olabilir.


# 11. Adım — Mini Sonuç Raporu

Bu bölümü GitHub README dosyana veya LinkedIn paylaşımına ekleyebilirsin.

## Kısa Rapor

Bu projede 100 satırlık örnek bir e-ticaret sipariş veri seti analiz edilmiştir.

Analiz sonucunda:

- Siparişlerin genel tutar seviyesi incelenmiştir.
- Tamamlanan siparişler üzerinden toplam ciro hesaplanmıştır.
- Üyelik tiplerine göre ortalama sipariş tutarı karşılaştırılmıştır.
- Kategori ve şehir bazlı ciro dağılımı analiz edilmiştir.
- Yüksek tutarlı siparişler filtrelenmiştir.
- İş dünyasına yönelik içgörüler çıkarılmıştır.

# 12. Adım — GitHub README Taslağı

Aşağıdaki metni GitHub'da `README.md` dosyası olarak kullanabilirsin.

```markdown
# E-Commerce Sales Analysis Project

## Project Overview

This project analyzes a sample e-commerce sales dataset using Python and Pandas.

The goal of the project is to understand order behavior, customer segments, revenue distribution, and business insights.

## Dataset

The dataset includes 100 sample e-commerce orders with the following columns:

- order id
- order date
- customer name
- city
- category
- product
- membership type
- payment method
- product price
- quantity
- discount rate
- total amount
- shipping duration
- order status

## Tools Used

- Python
- Pandas
- Google Colab

## Analysis Steps

1. Loaded CSV file into Google Colab
2. Explored the dataset with head(), shape, info(), and describe()
3. Calculated summary statistics
4. Filtered high-value orders
5. Analyzed customer segments by membership type
6. Compared revenue by category and city
7. Generated business insights

## Key Insights

- The average order amount is higher than the median, which suggests that some high-value orders increase the average.
- Completed orders generated a meaningful total revenue.
- Some categories contribute more revenue than others.
- Gold members tend to have higher average order amounts than standard members.
- Order values vary significantly, so customer segmentation can support better marketing decisions.

## Business Recommendations

- Focus retention campaigns on high-value customers.
- Analyze top-performing categories for growth opportunities.
- Create targeted campaigns for Gold and Silver members.
- Monitor high-value orders and shipping experience closely.

## Author

Bootcamp Final Project
```

# 13. Adım — GitHub'a Yükleme Rehberi

1. Colab dosyasını indir:
   - File > Download > Download .ipynb

2. GitHub'da yeni repository oluştur:
   - Repository adı önerisi: `ecommerce-sales-analysis`

3. Şu dosyaları yükle:
   - Notebook dosyası
   - CSV dosyası
   - README.md

4. README dosyasına:
   - Proje amacı
   - Kullanılan araçlar
   - Analiz adımları
   - İçgörüler
   - Öneriler

ekle.

5. LinkedIn'de projenin linkini paylaş.

# 14. Adım — LinkedIn Paylaşım Taslağı

Aşağıdaki metni kendine göre düzenleyerek LinkedIn'de paylaşabilirsin.

```text
İlk veri analizi projemi tamamladım! 🚀

Bu projede Python ve Pandas kullanarak örnek bir e-ticaret satış veri setini analiz ettim.

Çalışma kapsamında:

✅ CSV dosyasını Google Colab'e yükledim  
✅ Pandas ile veri setini inceledim  
✅ Özet istatistikleri yorumladım  
✅ Filtreleme ve segmentasyon yaptım  
✅ Kategori, şehir ve üyelik tipine göre analizler gerçekleştirdim  
✅ İş dünyasına yönelik içgörüler çıkardım  

Bu proje, veri analizi yolculuğumda öğrendiğim istatistik, Python ve Pandas konularını bir araya getirdiğim ilk portfolyo çalışmam oldu.

#Python #Pandas #DataAnalysis #DataAnalyst #GitHub
```

# Proje Tamamlandı 🎉

Bu notebook ile:

- CSV dosyası yükledin
- Pandas ile veri okudun
- EDA yaptın
- Özet istatistikleri yorumladın
- Filtreleme yaptın
- Segmentasyon yaptın
- İş sorularına cevap verdin
- GitHub'a yüklenebilir bir portfolyo projesi hazırladın